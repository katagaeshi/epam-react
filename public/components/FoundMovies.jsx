import React from 'react';
import PropTypes from 'prop-types';
import MovieTile from './MovieTile';
import '../styles/EmptyResults.css';

const extractMovies = movies => movies.map(movie => <MovieTile {...movie} />);

const FoundMovies = (props) => {
  const { movies } = props;
  const totalMessage = <span className={!movies ? 'EmptyResults' : null}>{props.total}</span>;

  return (
    <div>
      {totalMessage}
      {
        movies ?
          extractMovies(movies) :
          null
      }
    </div>
  );
};

FoundMovies.propTypes = {
  movies: PropTypes.arrayOf(PropTypes.string),
  total: PropTypes.string,
};

FoundMovies.defaultProps = {
  movies: null,
  total: 'No films found',
};

export default FoundMovies;
