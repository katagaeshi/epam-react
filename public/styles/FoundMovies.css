.FoundMovies {
  display: grid;
  grid-gap: 15px;
  grid-row-gap: 15px;
  grid-template-columns: repeat(auto-fit, minmax(220px, 1fr));
}
