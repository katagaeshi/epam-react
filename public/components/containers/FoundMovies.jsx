import { connect } from 'react-redux';
import FoundMovies from '../ui/FoundMovies';
import {
  setSortBy,
  sortMovies,
  setMovieDetails,
  setActivePanel,
  setRedirect,
} from '../../actions';

const mapStateToProps = state => (
  Object.assign({}, state.foundMovies.value, {
    option: state.sortBy,
    selectedMovie: state.movieDetails,
    activePanel: state.activePanel,
  })
);

const mapDispatchToProps = dispatch => ({
  onSortUpdate(changeEvent, movies) {
    const sortOption = changeEvent.target.value;
    dispatch(setSortBy(sortOption));
    dispatch(sortMovies(sortOption, movies));
  },
  onMovieClick(event) {
    if (event.target !== event.currentTarget) {
      dispatch(setMovieDetails(event.target.dataset.id));
      dispatch(setActivePanel('MovieDetails'));
      dispatch(setRedirect(true));
    }
    event.stopPropagation();
  },
});

const Container = connect(mapStateToProps, mapDispatchToProps)(FoundMovies);
export default Container;
