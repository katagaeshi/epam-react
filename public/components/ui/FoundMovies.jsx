import React from 'react';
import PropTypes from 'prop-types';
import MovieTile from './MovieTile';
import ResultsSort from './ResultsSort';
import '../../styles/EmptyResults.css';
import '../../styles/FoundMovies.css';

const extractMovies = (movies, onClick) => {
  const tiles = movies.map((movie, idx) => {
    const key = idx;
    return (<MovieTile
      idx={idx}
      key={key}
      {...movie}
    />);
  });
  return (
    <div
      onClick={onClick}
      role="presentation"
      className="FoundMovies"
    >
      {tiles}
    </div>
  );
};

const FoundMovies = (props) => {
  const {
    movies,
    option,
    onSortUpdate,
  } = props;

  let moviesFoundMessage = '';
  let resultsSort = null;
  if (props.total) {
    moviesFoundMessage = `${props.total} movies found`;
    const onUpdate = (event) => {
      onSortUpdate(event, movies);
    };
    resultsSort = (
      <ResultsSort
        onUpdate={onUpdate}
        checked={option}
      />
    );
  } else {
    moviesFoundMessage = FoundMovies.defaultProps.total;
  }
  const totalMessage = (
    <span
      className={!movies ? 'EmptyResults' : null}
    >
      {moviesFoundMessage}
    </span>
  );

  return (
    <div>
      {totalMessage}
      {resultsSort}
      {
        movies ?
          extractMovies(movies, props.onMovieClick) :
          null
      }
    </div>
  );
};

FoundMovies.propTypes = {
  movies: PropTypes.arrayOf(PropTypes.shape({
    src: PropTypes.string,
    title: PropTypes.string.isRequired,
    year: PropTypes.number,
    genre: PropTypes.string,
  })),
  total: PropTypes.string,
  option: PropTypes.string.isRequired,
  onSortUpdate: PropTypes.func.isRequired,
  onMovieClick: PropTypes.func.isRequired,
};

FoundMovies.defaultProps = {
  movies: null,
  total: 'No films found',
};

export default FoundMovies;
