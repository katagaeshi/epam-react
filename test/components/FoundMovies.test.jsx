import React from 'react';
import renderer from 'react-test-renderer';
import FoundMovies from './../../public/components/FoundMovies';

describe('FoundMovies', () => {
  it('Snapshot', () => {
    const tree = renderer
      .create(<FoundMovies />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
