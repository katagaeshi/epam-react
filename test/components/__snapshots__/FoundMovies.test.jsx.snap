// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FoundMovies Snapshot 1`] = `
<div>
  <span
    className="EmptyResults"
  >
    No films found
  </span>
</div>
`;
